io-engine/src/bdev/nexus/nexus_bdev.rs

```diff
@@ -1336,6 +1338,20 @@ pub async fn nexus_create_v2(
     }
 }
 
+async fn create_children_devices(
+    children: &[String],
+) -> Result<Vec<(String, String)>, Error> {
+    let mut children_devices = Vec::new();
+    for uri in children {
+        let device_name = device_create(uri).await.unwrap();
+        if device_lookup(&device_name).unwrap().is_zoned() && children.len() > 1 {
+            return Err(Error::ZonedReplicationNotImplemented {});
```

Comment:
If we are returning without creating all devices, don't we want to destroy back what we would've already created?